File: libNative/src/main/java/com/munch/lib/nativelib/helper/AppHelper.java
package com.munch.lib.nativelib.helper;

import android.app.Application;

/**
 * Created by Munch on 2018/12/17.
 */
public class AppHelper {

    private static Application mApplication;
    private static boolean isDebug = false;

    /**
     * @see #getApplication()
     */
    public static void init(Application application) {
        mApplication = application;
    }

    /**
     * @see #init(Application)
     */
    public static Application getApplication() {
        return mApplication;
    }

    public static boolean isDebug() {
        return isDebug;
    }


    public static void isDebug(boolean isDebug) {
        AppHelper.isDebug = isDebug;
    }
}

File: lib/src/main/java/com/munch/lib/app/AppHelper.kt
package com.munch.lib.app

import android.app.Application
import com.munch.lib.app.AppHelper.app
import com.munch.lib.app.AppHelper.init

/**
 * lib用于获取和管理application的方法类
 *
 * 如果使用到此类或者依赖此类的类或者方法，则必须在之前调用[init]方法进行初始化
 *
 * @see init
 * @see app
 *
 * Create by munch1182 on 2021/8/5 13:56.
 */
object AppHelper {

    private lateinit var application: Application

    val app: Application
        get() = application

    fun init(app: Application) {
        application = app
    }

}